### PR TITLE
postgres-cdc checkpointing: fix LSN parsing bug + refactor for efficiency 

### PR DESCRIPTION
**Pull request by sergio-ropero**
Issue : https://github.com/airbytehq/airbyte/issues/21009

## Test
Performed a huge sync to get a LSN larger than Integer:
Screenshot 2023-03-27 at 19 07 54

Checking the amount of CDC state messages sent (165+final) in a local sync for 3M rows:
```
2023-03-27 18:08:34 [32mINFO[m i.a.w.g.DefaultReplicationWorker(getReplicationOutput):538 - sync summary: {
  "status" : "completed",
  "recordsSynced" : 3000000,
  "bytesSynced" : 468000000,
  "startTime" : 1679940332667,
  "endTime" : 1679940514626,
  "totalStats" : {
    "bytesEmitted" : 468000000,
    "destinationStateMessagesEmitted" : 166,
    "destinationWriteEndTime" : 1679940514625,
    "destinationWriteStartTime" : 1679940332772,
    "meanSecondsBeforeSourceStateMessageEmitted" : 8,
    "maxSecondsBeforeSourceStateMessageEmitted" : 8,
    "maxSecondsBetweenStateMessageEmittedandCommitted" : 1,
    "meanSecondsBetweenStateMessageEmittedandCommitted" : 0,
    "recordsEmitted" : 3000000,
    "recordsCommitted" : 3000000,
    "replicationEndTime" : 1679940514626,
    "replicationStartTime" : 1679940332667,
    "sourceReadEndTime" : 1679940513851,
    "sourceReadStartTime" : 1679940332714,
    "sourceStateMessagesEmitted" : 166
  },
  "streamStats" : [ {
    "streamName" : "test_table",
    "stats" : {
      "bytesEmitted" : 468000000,
      "recordsEmitted" : 3000000,
      "recordsCommitted" : 3000000
    }
  } ]
}
```
Example:
Screenshot 2023-03-27 at 20 25 41

Screenshot 2023-03-27 at 20 29 20

### CODE REVIEW

**Comment by glevineLeap** [comment]
Does this change resolve https://github.com/airbytehq/airbyte/issues/24226?

**Comment by glevineLeap** [comment]
> > Does this change resolve #24226?
> 
> Hello @glevineLeap, It is correct, this PR is meant to fix that issue, although for now it should not be happening in version 2.0.11

Thanks @sergio-ropero. I'm not fully familiar with the release process, so I wanted to clarify: will this fix be released to affected connectors as they pass or will it not be released to any until all affected connectors pass and PR is merged?

**Comment by rodireich** [comment]
This was the change we had to roll back because of some race condition with us writing to the file while debezium reading from it, right?
Can you point where the fix for the race condition is?

**Comment by subodh1810** [comment]
> This was the change we had to roll back because of some race condition with us writing to the file while debezium reading from it, right? Can you point where the fix for the race condition is?

Yup! Look at the changes in the file `AirbyteFileOffsetBackingStore` and `DebeziumStateDecoratingIterator` where we swallow the `ConnectException`  exception

File: airbyte-integrations/bases/debezium/src/main/java/io/airbyte/integrations/debezium/internals/AirbyteFileOffsetBackingStore.java [hunk 1]
```diff
@@ -108,34 +108,37 @@ private static ByteBuffer stringToByteBuffer(final String s) {
...
+      throw new ConnectException("Expected HashMap but found " + obj.getClass());
+    final Map<byte[], byte[]> raw = (Map<byte[], byte[]>) obj;
+    final Map<ByteBuffer, ByteBuffer> data = new HashMap<>();
+    for (final Map.Entry<byte[], byte[]> mapEntry : raw.entrySet()) {
+      final ByteBuffer key = (mapEntry.getKey() != null) ? ByteBuffer.wrap(mapEntry.getKey()) : null;
+      final ByteBuffer value = (mapEntry.getValue() != null) ? ByteBuffer.wrap(mapEntry.getValue()) : null;
+      data.put(key, value);
+    }
+
+    return data;
```
[the remark this answers is not in the document]
Comment (akashkulk) [reply airbyte-integrations/bases/debezium/src/main/java/io/airbyte/integrations/debezium/internals/AirbyteFileOffsetBackingStore.java:141]: +1 - would be curious to know

[the remark this answers is not in the document]
Comment (subodh1810) [reply airbyte-integrations/bases/debezium/src/main/java/io/airbyte/integrations/debezium/internals/AirbyteFileOffsetBackingStore.java:141]: @rodireich @akashkulk Sergio had mentioned that moving out of the try-catch block everything that is not related with IO operations (type checks, in memory object loops, ...) frees the resource and, if it got write before closing, we don't face a java.io.StreamCorruptedException: unexpected EOF in middle of data block exception.

**Review by akashkulk — APPROVED** [review]

**Comment by subodh1810** [comment]
/publish connector=connectors/source-mssql-strict-encrypt run-tests=false

**Pull request merged**